graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/dml/InsertMutationFetcher.java

```diff
@@ -41,34 +40,31 @@ public InsertMutationFetcher(
   }
 
   @Override
-  protected String buildStatement(DataFetchingEnvironment environment, DataStore dataStore) {
-    Insert insert =
-        QueryBuilder.insertInto(keyspaceId, tableId).valuesByIds(buildInsertValues(environment));
+  protected BoundQuery buildQuery(DataFetchingEnvironment environment, DataStore dataStore) {
+    boolean ifNotExists =
+        environment.containsArgument("ifNotExists")
+            && environment.getArgument("ifNotExists") != null
+            && (Boolean) environment.getArgument("ifNotExists");
 
-    if (environment.containsArgument("ifNotExists")
-        && environment.getArgument("ifNotExists") != null
-        && (Boolean) environment.getArgument("ifNotExists")) {
-      insert = insert.ifNotExists();
-    }
-    if (environment.containsArgument("options") && environment.getArgument("options") != null) {
-      Map<String, Object> options = environment.getArgument("options");
-      if (options.containsKey("ttl") && options.get("ttl") != null) {
-        insert = insert.usingTtl((Integer) options.get("ttl"));
-      }
-    }
-
-    return insert.asCql();
+    return dataStore
+        .queryBuilder()
+        .insertInto(table.keyspace(), table.name())
+        .value(buildInsertValues(environment))
+        .ifNotExists(ifNotExists)
+        .ttl(getTTL(environment))
+        .build()
+        .bind();
```

Comment:
What about a mode where the values are inlined in the query (like right now), but when the datastore executes it, it replaces all the values by bind markers at the last moment, and prepares a parameterized query?

The problem with the current approach is that if I want to switch to prepared statements, I have to keep track of all the values until I'm ready to call `bind()`. That's an extra `List<Object>` that needs to be passed around into every method.

Reply:
I'll check what's wrong (I'm surprised by what you observe since there is unit tests but ...), but it is supposed to (always) work as you describe it in your last comment.

That is, the built query are supposed to always use bound values internally no matter what and to never ever inline a value in the query string (though, for the convenience of debugging, the `BuiltQuery#toString` method does return a string with inlined values; it's just that this string shouldn't be used for actual querying. I might have messed that up somewhere, I'l check).

Reply:
Right, I think I missed something: I just checked `system.prepared_statements` and the query does indeed have bind markers.